### PR TITLE
implemented parse_link_header

### PR DESCRIPTION
**Pull request by gsheng0**

### CODE REVIEW

File: src/fetcher.py [hunk 1]
```diff
@@ -64,10 +64,14 @@ def __fetch_paginated(self, url: str) -> list[dict]:
 
 def parse_link_header(link_header: str) -> dict[str, str]:
     """Parse the `Link` header from a GitHub API response into a dictionary"""
-    # TODO: implement
-    return {
-        "last": "https://api.github.com/repositories/198065251/commits?per_page=1&page=3095"
-    }
+    chunks = link_header.split(",")
+    output = {}
+    for i in range(0, len(chunks)):
```
Comment (dyc3) [inline src/fetcher.py:69]: nit: this would be more readable
```suggestion
    for chunk in chunks:
```

File: src/fetcher.py [hunk 2]
```diff
@@ -64,10 +64,14 @@ def __fetch_paginated(self, url: str) -> list[dict]:
...
+        link = chunks[i][chunks[i].index("<") + 1 : chunks[i].index(">")]
+        name = chunks[i][chunks[i].index("\"") + 1 : chunks[i].index("\"", chunks[i].index("\"") + 1)]
```
Comment (dyc3) [inline src/fetcher.py]: In my experience, string parsing like this tends to be extremely brittle. This would be easier to maintain using a regex instead.

**Review by dyc3 — APPROVED** [review]
This is fine. I'll make a PR with tests

File: src/fetcher.py (hunk 2 above)
Comment (gsheng0) [reply src/fetcher.py]: Ah, I hadn't thought of using regex for something like this. I'll try that next time

**Pull request merged**